### PR TITLE
Fix all remaining build/include warnings

### LINKED ISSUE #2364: cpplint failing due to [build/include] warnings
`[build/include]` is disabled in `CPPLINT.cfg` pending the resolution of this issue.

```
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/MathematicalProgram.cpp:0: warning: drake/solvers/MathematicalProgram.cpp should include its header file drake/solvers/MathematicalProgram.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/MobyLCP.cpp:6: warning: drake/solvers/MobyLCP.cpp should include its header file drake/solvers/MobyLCP.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/qpSpline/ContinuityConstraint.cpp:2: warning: drake/solvers/qpSpline/ContinuityConstraint.cpp should include its header file drake/solvers/qpSpline/ContinuityConstraint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/qpSpline/SplineInformation.cpp:2: warning: drake/solvers/qpSpline/SplineInformation.cpp should include its header file drake/solvers/qpSpline/SplineInformation.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/solvers/qpSpline/ValueConstraint.cpp:2: warning: drake/solvers/qpSpline/ValueConstraint.cpp should include its header file drake/solvers/qpSpline/ValueConstraint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/controllers/InstantaneousQPController.cpp:2: warning: drake/systems/controllers/InstantaneousQPController.cpp should include its header file drake/systems/controllers/InstantaneousQPController.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/collision/Element.cpp:3: warning: drake/systems/plants/collision/Element.cpp should include its header file drake/systems/plants/collision/Element.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/collision/Model.cpp:3: warning: drake/systems/plants/collision/Model.cpp should include its header file drake/systems/plants/collision/Model.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/constraint/constructPtrRigidBodyConstraint.cpp:1: warning: drake/systems/plants/constraint/constructPtrRigidBodyConstraint.cpp should include its header file drake/systems/plants/constraint/constructPtrRigidBodyConstraint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/HelicalJoint.cpp:0: warning: drake/systems/plants/joints/HelicalJoint.cpp should include its header file drake/systems/plants/joints/HelicalJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/PrismaticJoint.cpp:0: warning: drake/systems/plants/joints/PrismaticJoint.cpp should include its header file drake/systems/plants/joints/PrismaticJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/QuaternionFloatingJoint.cpp:2: warning: drake/systems/plants/joints/QuaternionFloatingJoint.cpp should include its header file drake/systems/plants/joints/QuaternionFloatingJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/RevoluteJoint.cpp:0: warning: drake/systems/plants/joints/RevoluteJoint.cpp should include its header file drake/systems/plants/joints/RevoluteJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/joints/RollPitchYawFloatingJoint.cpp:2: warning: drake/systems/plants/joints/RollPitchYawFloatingJoint.cpp should include its header file drake/systems/plants/joints/RollPitchYawFloatingJoint.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/RigidBody.cpp:4: warning: drake/systems/plants/RigidBody.cpp should include its header file drake/systems/plants/RigidBody.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/rigidBodyTreeMexFunctions.cpp:1: warning: drake/systems/plants/rigidBodyTreeMexFunctions.cpp should include its header file drake/systems/plants/rigidBodyTreeMexFunctions.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/shapes/Element.cpp:0: warning: drake/systems/plants/shapes/Element.cpp should include its header file drake/systems/plants/shapes/Element.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/shapes/Geometry.cpp:1: warning: drake/systems/plants/shapes/Geometry.cpp should include its header file drake/systems/plants/shapes/Geometry.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/shapes/VisualElement.cpp:0: warning: drake/systems/plants/shapes/VisualElement.cpp should include its header file drake/systems/plants/shapes/VisualElement.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/plants/xmlUtil.cpp:2: warning: drake/systems/plants/xmlUtil.cpp should include its header file drake/systems/plants/xmlUtil.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/robotInterfaces/BodyMotionData.cpp:0: warning: drake/systems/robotInterfaces/BodyMotionData.cpp should include its header file drake/systems/robotInterfaces/BodyMotionData.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/systems/trajectories/PiecewiseFunction.cpp:2: warning: drake/systems/trajectories/PiecewiseFunction.cpp should include its header file drake/systems/trajectories/PiecewiseFunction.h  [build/include] [5]
/home/ubuntu/workspace/linux-cpplint-continuous/drake/util/yaml/yamlUtil.cpp:0: warning: drake/util/yaml/yamlUtil.cpp should include its header file drake/util/yaml/yamlUtil.h  [build/include] [5]
```

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
This PR builds upon #3974 and fixes #2364.  This is the final 5% of interesting fixes and the global warning enable toggle.  The commits are curated in case reviewers want to explore detail-by-detail.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/3980)

### CODE REVIEW

**Comment by david-german-tri** [comment]
:lgtm:

---

Review status: 0 of 21 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-:-KVV6zga69deGAFN6uSK:bnfp4nl)*

**Comment by david-german-tri** [comment]
+@david-german-tri 

---

Reviewed 21 of 21 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-:-KVV70iD3ewB69yvLD6H:bqmdn1l)*

**Comment by mwoehlke-kitware** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-KVVG1HVDDwsj97T4ijt:-KVVG1HXTOULVyjEwGQo:bkm8fy2) ([raw file](https://github.com/robotlocomotion/drake/blob/6f3fb2c1c21d53f6a8dc37fecbf31cea09f6ad8b/drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp#L4)):*
> ```cpp
> #include <bot_lcmgl_client/lcmgl.h>
> 
> #include "lcmtypes/drake_lcmt_zmp_com_observer_state.h"
> ```

Um... this doesn't seem right. We don't use `lcmtypes/` anywhere else (as of @8094948fb80a, the only instance of `lcmtypes/` is a JAR); it is part of the interface include directories of the LCM types library. Is this linked to `drakeLCMTypes`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980)*

**Comment by jwnimmer-tri** [comment]
*[drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-KVVG1HVDDwsj97T4ijt:-KVVHyusnk4zQGL11w-w:b-dinlfd) ([raw file](https://github.com/robotlocomotion/drake/blob/6f3fb2c1c21d53f6a8dc37fecbf31cea09f6ad8b/drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp#L4)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary>
> Um... this doesn't seem right. We don't use `lcmtypes/` anywhere else (as of @8094948fb80a, the only instance of `lcmtypes/` is a JAR); it is part of the interface include directories of the LCM types library. Is this linked to `drakeLCMTypes`?
</details>
Okay.  What is the correct include path for this file?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980)*

**Comment by mwoehlke-kitware** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-KVVG1HVDDwsj97T4ijt:-KVVP8F3ADR0ZC5_uTF1:b8hu97a) ([raw file](https://github.com/robotlocomotion/drake/blob/6f3fb2c1c21d53f6a8dc37fecbf31cea09f6ad8b/drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp#L4)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> Okay.  What is the correct include path for this file?
</details>
`.`... basically, I don't understand/trust why this change is necessary

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980)*

**Comment by jwnimmer-tri** [comment]
*[drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-KVVG1HVDDwsj97T4ijt:-KVVQ8VjNsHZNQGfiOrH:b-q8w2zf) ([raw file](https://github.com/robotlocomotion/drake/blob/6f3fb2c1c21d53f6a8dc37fecbf31cea09f6ad8b/drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp#L4)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary>
> `.`... basically, I don't understand/trust why this change is necessary
</details>
Cpplint is like "Wow, do you _really_ mean to include a file from within your _own project_ using no subdirectory name?" and I'm like  "Huh, yeah, that looks stupid, I wonder what the correct path is?" and I `find -name drake_lcmt_zmp_com_observer_state.h` in my build directory and it shows `drake/lcmtypes/drake_lcmt_zmp_com_observer_state.h` so I'm like cool man, I'll add `lcmtypes/` here and then Jenkins is like "No way may, only half of your builds are going to pass now" and I'm like "WTF".

Bottom line, if you tell me the correct include path, I'll change this line to match.  If this answer is "no subdirectory at all" I will also inform cpplint of that fact.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980)*

**Comment by mwoehlke-kitware** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-KVVG1HVDDwsj97T4ijt:-KVVWo2BCLAf7CGlJEJ1:b-s4borc) ([raw file](https://github.com/robotlocomotion/drake/blob/6f3fb2c1c21d53f6a8dc37fecbf31cea09f6ad8b/drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp#L4)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> Cpplint is like "Wow, do you _really_ mean to include a file from within your _own project_ using no subdirectory name?" and I'm like  "Huh, yeah, that looks stupid, I wonder what the correct path is?" and I `find -name drake_lcmt_zmp_com_observer_state.h` in my build directory and it shows `drake/lcmtypes/drake_lcmt_zmp_com_observer_state.h` so I'm like cool man, I'll add `lcmtypes/` here and then Jenkins is like "No way may, only half of your builds are going to pass now" and I'm like "WTF".
> 
> Bottom line, if you tell me the correct include path, I'll change this line to match.  If this answer is "no subdirectory at all" I will also inform cpplint of that fact.
</details>
For a C LCM header, `.` (i.e. no subdirectory) is indeed right... I guess this is the only time we use a C LCM type instead of C++. (For C++ LCM types, it is `<package>/<type>.hpp`, which I guess is why cpplint doesn't complain.)

Also, s/drakeLCMTypes/drake_lcmtypes/ in my previous comment; I wasn't paying attention that this is a C binding.

Alternatively, port this code to use the C++ type? :wink:

As a third option, `drake/lcmtypes` might work; I'd be okay with that. We dropped `lcmtypes/` originally for consistency, because the types didn't use it internally. If you don't mind that inconsistency here, I think `drake/lcmtypes` is perhaps more consistent with how we generally `#include` headers. (We could also revisit making the generated files prefix their internal includes...)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980)*

**Comment by jwnimmer-tri** [comment]
Review status: 17 of 20 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-KVVG1HVDDwsj97T4ijt:-KVVdKH_7mHAy5bw4vWZ:bogjgoi) ([raw file](https://github.com/robotlocomotion/drake/blob/6f3fb2c1c21d53f6a8dc37fecbf31cea09f6ad8b/drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp#L4)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary>
> For a C LCM header, `.` (i.e. no subdirectory) is indeed right... I guess this is the only time we use a C LCM type instead of C++. (For C++ LCM types, it is `<package>/<type>.hpp`, which I guess is why cpplint doesn't complain.)
> 
> Also, s/drakeLCMTypes/drake_lcmtypes/ in my previous comment; I wasn't paying attention that this is a C binding.
> 
> Alternatively, port this code to use the C++ type? :wink:
> 
> As a third option, `drake/lcmtypes` might work; I'd be okay with that. We dropped `lcmtypes/` originally for consistency, because the types didn't use it internally. If you don't mind that inconsistency here, I think `drake/lcmtypes` is perhaps more consistent with how we generally `#include` headers. (We could also revisit making the generated files prefix their internal includes...)
</details>
Ah.  I also didn't realize it was plain C.  I've just NOLINT'd the beehive, and will back away slowly.  Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980)*

**Comment by jwnimmer-tri** [comment]
@david-german-tri @mwoehlke-kitware PTAL.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-:-KVVltL1RgvX6Ykk7Vc8:b-bs73b9)*

**Comment by mwoehlke-kitware** [comment]
:lgtm:

---

Review status: 17 of 20 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-KVVG1HVDDwsj97T4ijt:-KVW8QNvnvLbrYekTgrq:b-9qlhic) ([raw file](https://github.com/robotlocomotion/drake/blob/6f3fb2c1c21d53f6a8dc37fecbf31cea09f6ad8b/drake/systems/controllers/zmpCoMObserverStateVisualizer.cpp#L4)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary>
> Ah.  I also didn't realize it was plain C.  I've just NOLINT'd the beehive, and will back away slowly.  Done.
</details>
Hmm... yes, on second thought, `drake/<type>.hpp` might be preferred to `drake/lcmtypes/drake/<type>.hpp`, in which case, yes, "C code is odd" :smile:. (In particular, LCM believes that C headers don't go in a package-name subdirectory, but instead *start with* the package name. Which, on further reflection, does seem to imply intent of a flat hierarchy.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3980#-:-KVW9K293A6BrXHDKEt7:bnfp4nl)*
